### PR TITLE
Service entity methods

### PR DESCRIPTION
**Pull request by Zarathustra4**

### CODE REVIEW

File: src/main/java/com/passpnu/passwordmanager/controller/ServiceController.java [hunk 1]
```diff
@@ -1,4 +1,58 @@
...
+        return serviceEntityService.getServiceList();
+    }
+
+    @PutMapping
+    public ResponseEntity<String> editService(@RequestBody ServiceDto service) throws NameNotFoundException {
+        if( !serviceEntityService.existsByDomain(service) ){
+            return new ResponseEntity<>("There are not such a service", HttpStatus.BAD_REQUEST);
+        }
+
+        Optional<ServiceDto> serviceDto = Optional.ofNullable(serviceEntityService.putService(service));
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/controller/ServiceController.java]: Wrapping object in `Optional` just for null-check is a really expensive operation. A regular null-check will do the trick

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java [hunk 2]
```diff
@@ -1,7 +1,95 @@
...
+import java.util.List;
+import java.util.Optional;
+
 @Service
-public class ServiceEntityServiceImpl {
+@AllArgsConstructor
+public class ServiceEntityServiceImpl implements ServiceEntityService{
+    ServiceRepository serviceRepository;
+
+    private ServiceEntity dtoToEntity(ServiceDto serviceDto){
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java]: This is not an expected functionality of service. You should create special code-units (mappers) for that, better with Mapstruct

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java [hunk 3]
```diff
@@ -1,7 +1,95 @@
...
+                service.getDomain()
+        );
+
+        if(serviceOptional.isEmpty()){
+            throw new NameNotFoundException("The service is not found");
+        }
+
+        ServiceEntity serviceEntity = serviceOptional.get();
+
+        serviceEntity.setDescription(service.getDescription());
```
Comment (justme1135) [inline src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java:53]: Same here

**Review by justme1135 — CHANGES_REQUESTED** [review]

File: src/main/java/com/passpnu/passwordmanager/controller/ServiceController.java (hunk 1 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/controller/ServiceController.java]: fixed

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java (hunk 2 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java]: Mapper is added

File: src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java (hunk 3 above)
Comment (Zarathustra4) [reply src/main/java/com/passpnu/passwordmanager/service/ServiceEntityServiceImpl.java:53]: fixed

**Pull request merged**